modules/environment-roles/templates/terraform_scripts_policy.json.tpl

```diff
@@ -0,0 +1,119 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Action": [
+        "dynamodb:DeleteItem",
+        "dynamodb:GetItem",
+        "dynamodb:PutItem",
+        "iam:AddRoleToInstanceProfile",
+        "iam:AttachRolePolicy",
+        "iam:CreateInstanceProfile",
+        "iam:CreatePolicy",
+        "iam:CreatePolicyVersion",
+        "iam:CreateRole",
+        "iam:DeleteInstanceProfile",
+        "iam:DeletePolicy",
+        "iam:DeletePolicyVersion",
+        "iam:DeleteRole",
+        "iam:DetachRolePolicy",
+        "iam:GetInstanceProfile",
+        "iam:GetPolicy",
+        "iam:GetPolicyVersion",
+        "iam:GetRole",
+        "iam:ListAttachedRolePolicies",
+        "iam:ListInstanceProfilesForRole",
+        "iam:ListPolicyVersions",
+        "iam:PassRole",
+        "iam:RemoveRoleFromInstanceProfile",
+        "iam:TagRole",
+        "kms:CreateAlias",
+        "kms:CreateGrant",
+        "kms:Decrypt",
+        "kms:DeleteAlias",
+        "kms:DescribeKey",
+        "kms:DescribeKey*",
+        "kms:EnableKey",
+        "kms:EnableKeyRotation",
+        "kms:Encrypt",
+        "kms:GenerateDataKey*",
+        "kms:GetKeyPolicy",
+        "kms:GetKeyRotationStatus",
+        "kms:ListGrants",
+        "kms:ListKeyPolicies",
+        "kms:ListResourceTags",
+        "kms:ListRetirableGrants",
+        "kms:PutKeyPolicy",
+        "kms:ReEncrypt*",
+        "kms:RevokeGrant",
+        "kms:ScheduleKeyDeletion",
+        "kms:TagResource",
+        "kms:UntagResource",
+        "kms:UpdateAlias",
+        "kms:UpdateKeyDescription",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:DeleteLogGroup",
+        "logs:DeleteLogStream",
+        "logs:DescribeLogGroups",
+        "logs:DescribeLogStreams",
+        "logs:ListTagsLogGroup",
+        "logs:PutRetentionPolicy",
+        "logs:TagLogGroup",
+        "s3:GetObject",
+        "s3:PutObject",
+        "ssm:GetParameter"
+      ],
+      "Resource": [
+        "arn:aws:dynamodb:eu-west-2:${account_id}:table/tdr-terraform-state-lock-scripts",
+        "arn:aws:iam::${account_id}:instance-profile/bastion",
+        "arn:aws:iam::${account_id}:role/BastionEC2RoleIntg",
+        "arn:aws:kms:eu-west-2:${account_id}:alias/tdr-bastion-encryption-intg",
+        "arn:aws:kms:eu-west-2:${account_id}:key/*",
+        "arn:aws:s3:::tdr-terraform-state-scripts/*",
+        "arn:aws:ssm:eu-west-2:${account_id}:parameter/intg/consignmentapi/database/password",
```

Comment:
Should the TDR env be hardcoded: "... :parameter/**intg**/consignmentapi/..."

Reply:
No it definitely shouldn't be. I've passed in the environment instead.